### ISSUE TITLE
Shell 'CurrentPage' property does not have a value to bind it at the first load. 

### ISSUE THREAD

**Issue by EstebanP-dev** [dotnet/maui#21200]
### Description

Hi, devs. I’m attempting to implement a global Title View without a setter in a Control Template or similar. Therefore, I’ve decided to use the bindable properties of Shell to recreate this. However, I’ve noticed that the value isn’t loading correctly. Here are some notes about:

- The title isn’t displaying.
- The value that the binding sends to any converter is null.

This is the page on its first load:
![Screenshot_1710393715](https://github.com/dotnet/maui/assets/71358592/c8b6c31d-df4a-481c-94fb-d86cb1ea52dd)

This is after hot reload:
![Screenshot_1710393792](https://github.com/dotnet/maui/assets/71358592/d1e50906-b35c-4dd6-b283-43bbfb98c817)

Please let me know if I was unclear or incorrect.

I'm using the 8.0.10 version.

Thanks a lot!

### Steps to Reproduce

UI:
1. Set up a title in Main Page.
2. Add Title View at AppShell binding the CurrentPage property of itself also with a property called Title.

Converter:
 1. Set up a title in Main Page.
 2. Create a convertion validating the value paramater if it is a page type.
 3. Add Title View at AppShell binding the CurrentPage property of itself.
 4. Insert the converter in the bindging brackets.

### Link to public reproduction project repository

https://github.com/EstebanP-dev/OpenReportBugs/tree/maui-shell-binding-currentpage-on-titleview

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

**Comment by kevinxufei** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 2. Can repro on Android platform with sample project.

**Comment by PureWeen** [comment]
@EstebanP-dev
Can you use the Navigated event on Shell to trigger what you need? Once that fires CurrentPage should be set

You could fire OnpropertyChanged(<CurrentPage>) inside Navigated

**Comment by EstebanP-dev** [comment]
Yeah, with the OnNavigated, it works. However, wouldn't it be better to use the property binding? It could be with bindable property with OnWaySource binding mode. @PureWeen
